src/program.js

```diff
@@ -569,29 +569,31 @@ Example: $0 --help run.
           describe:
             'Use a proxy to access the signing API. ' +
             'Example: https://yourproxy:6000 ',
-          demandOption: false,
           type: 'string',
         },
         'use-submission-api': {
           describe:
             'Sign using the addon submission API',
-          demandOption: false,
           type: 'boolean',
         },
         'id': {
           describe:
             'A custom ID for the extension. This has no effect if the ' +
             'extension already declares an explicit ID in its manifest.',
-          demandOption: false,
           type: 'string',
         },
         'timeout': {
           describe: 'Number of milliseconds to wait before giving up',
           type: 'number',
         },
         'channel': {
-          describe: 'The channel for which to sign the addon. Either ' +
-          '\'listed\' or \'unlisted\'',
+          describe: 'The channel for which to sign the addon.',
+          type: 'string',
+          choices: ['listed', 'unlisted'],
+        },
+        'metadata': {
+          describe: 'A JSON object containing metadata to be passed to the ' +
+          'API.  Only used with `use-submission-api`',
           type: 'string',
```

Comment:
`metadata` feels a bit generic to me, even in the context of the sign command, I'm wondering if `listing-metadata` or `store-metadata` may be more immediately clear about what metadata that refers to.

Reply:
We don't refer to AMO as a `store` anywhere, so not `store-metadata`.  I guess `listing-metadata` is almost accurate - though there are a few properties you can set/update that aren't listing related (`approval_notes` and `source` - though we don't support uploads for `source`, nor editing the version to add it)

Reply:
An alternative may be `addon-metadata` but I'm not sure if that may be clear enough or likely be confused for something related to the manifest file (like the browser specific settings), if we believe that `listing-metadata` is accurate enough (even if technically a couple of those are not part of what a listing is on the AMO side), I'd be ok with going for `listing-metadata`.

Reply:
@eviljeff I mentioned this to @willdurand (while we have been discussed about also other things that I wanted to gather his perspective for once he was back) and we both agreed that your proposed name `amo-metadata` sounds even better (and it is also shorter, which is also nice) and so let's go for `amo-metadata` as you also suggested.

----

As a side note (I'm mentioning here mainly because it is also related to the submit-addon module and about using the same "option name convention" being chosen here):

@willdurand and I also discussed about that remaining `api-host` vs `api-url-prefix` detail form the initial pull request for the submit-addon.js module, and we agreed on some changes, and one was to rename the new option `amo-base-url`, which would also make the two new options to nicely share `amo-` in their option name.
But let's discuss between the three of us (you, @willdurand and I) in more detail about changes related to the `api-host` option name tomorrow.